### PR TITLE
Add and update tests

### PR DESCRIPTION
**Pull request by VladimirWork**

### CODE REVIEW

File: system/indy-node-tests/TestAdHocSuite.py [hunk 1]
```diff
@@ -52,7 +51,8 @@ class TestAdHocSuite:
             assert res3['result']['seqNo'] is not None
 
             # check that pool is ok
-            await ensure_pool_is_in_sync(nodes_num=nodes_num)
+            await ensure_all_nodes_online(pool_handler, wallet_handler, trustee_did)
+            await ensure_ledgers_are_in_sync(pool_handler, wallet_handler, trustee_did)
             await ensure_state_root_hashes_are_in_sync(pool_handler, wallet_handler, trustee_did)
```
Comment (skhoroshavin) [inline system/indy-node-tests/TestAdHocSuite.py]: I see these functions together quite often, why not create wrapper like `ensure_pool_is_okay`?

Comment (VladimirWork) [reply system/indy-node-tests/TestAdHocSuite.py]: Done.

File: system/draft/test_misc.py [hunk 2]
```diff
@@ -2900,34 +2900,141 @@ def test_misc_aws_demotion_promotion():
...
+    node_to_demote = choice(node_list)
+    # demote it
+    demote_tasks = []
+    for i in range(demote_count):
+        task = demote_node(pool_handler, wallet_handler, trustee_did, node_to_demote, pool_info[node_to_demote])
+        demote_tasks.append(task)
+    await asyncio.gather(*demote_tasks, return_exceptions=True)
+    await pool.refresh_pool_ledger(pool_handler)
+    # make sure VC is done
+    new_primary = await ensure_primary_changed(pool_handler, wallet_handler, trustee_did, primary)
```
[the remark this answers is not in the document]
Comment (VladimirWork) [reply system/draft/test_misc.py]: Done.

File: system/draft/test_misc.py [hunk 3]
```diff
@@ -2900,34 +2900,141 @@ def test_misc_aws_demotion_promotion():
...
+@pytest.mark.asyncio
+async def test_misc_cyclic_demotions_promotions(
+        docker_setup_and_teardown, pool_handler, wallet_handler, get_default_trustee, check_no_failures_fixture,
+        nodes_num, iterations, nyms_count
+):
+    trustee_did, _ = get_default_trustee
+    pool_info = get_pool_info('1')
+    node_list = ['Node{}'.format(x) for x in range(1, nodes_num + 1)]
+
+    for i in range(iterations):
```
[the remark this answers is not in the document]
Comment (VladimirWork) [reply system/draft/test_misc.py]: Done.

File: system/draft/test_misc.py [hunk 4]
```diff
@@ -2900,34 +2900,141 @@ def test_misc_aws_demotion_promotion():
...
+                _data = {
+                    'alias': req_data['node_alias'],
+                    'services': ['VALIDATOR']
+                }
+                try:
+                    req = await ledger.build_node_request(trustee_did, req_data['node_dest'], json.dumps(_data))
+                    await ledger.sign_and_submit_request(pool_handle, wallet_handle, trustee_did, req)
+                    # start promoted node
+                    host = testinfra.get_host('ssh://persistent_node' + req_data['node_alias'][4:])
+                    host.run('sudo systemctl start indy-node')
```
[the remark this answers is not in the document]
Comment (VladimirWork) [reply system/draft/test_misc.py]: We need determined behaviour here.

File: system/draft/test_misc.py [hunk 5]
```diff
@@ -2900,34 +2900,141 @@ def test_misc_aws_demotion_promotion():
...
-            try:
-                await demote_node(
-                    pool_handle, wallet_handle, trustee_did, req_data['node_alias'], req_data['node_dest']
-                )
-            except PoolLedgerTimeout:
-                await asyncio.sleep(interval)
-                continue
+            req_data_list = []
+            for i in range(nodes_num):
+                req_data_list.append(random.choice(pool_data[:-1]))  # keep 25th node always in pool
```
[the remark this answers is not in the document]
Comment (VladimirWork) [reply system/draft/test_misc.py]: Done.

**Pull request merged**